R/module_nested_tabs.R

```diff
@@ -330,11 +327,8 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, is_module_specifi
 #' @keywords internal
 #'
 calculate_hashes <- function(datanames, datasets) {
-  sapply(
-    datanames,
-    simplify = FALSE,
-    function(x) {
-      rlang::hash(datasets$get_data(x, filtered = FALSE))
-    }
-  )
+  if (is.null(names(datanames))) {
+    datanames <- structure(datanames, names = datanames)
+  }
+  lapply(datanames, function(x) rlang::hash(datasets$get_data(x, filtered = FALSE)))
```

Comment:
```suggestion
  lapply(datanames, function(x) rlang::hash(datasets$get_data(x, filtered = FALSE)))
```

Dropping names from the list will cause dropping hashes from the code. See here
https://github.com/insightsengineering/teal/blob/720a374771b296460f0eee9dabf00d4891f7c831/R/get_rcode_utils.R#L73